src/components/ExecutionDetailsSheet.tsx

```diff
@@ -1,47 +1,166 @@
+import type { Content } from "@radix-ui/react-dialog";
+import { useQuery } from "@tanstack/react-query";
+import { useLocation } from "@tanstack/react-router";
 import { ArrowDownToLine, InfoIcon, Terminal } from "lucide-react";
+import { type ComponentProps, useState } from "react";
 
+import type {
+  GetArtifactsApiExecutionsIdArtifactsGetResponse,
+  GetExecutionInfoResponse,
+} from "@/api/types.gen";
 import { Button } from "@/components/ui/button";
-import { ScrollArea } from "@/components/ui/scroll-area";
-import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
-import type { TaskSpec } from "@/componentSpec";
-
 import {
   Sheet,
   SheetContent,
   SheetHeader,
   SheetTitle,
   SheetTrigger,
-} from "./ui/sheet";
-
-interface ExecutionDetailsSheetProps {
-  taskSpec: TaskSpec;
-  taskId: string;
-  runStatus?: string;
-}
-
-const ExecutionDetailsSheet = ({
-  taskSpec,
-  taskId,
-  runStatus,
-}: ExecutionDetailsSheetProps) => {
+} from "@/components/ui/sheet";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import { cn } from "@/lib/utils";
+import { type RunDetailParams, runDetailRoute } from "@/router";
+import { API_URL, RUNS_BASE_PATH } from "@/utils/constants";
+
+const formatLogLine = (line: string) => {
+  const timestampMatch = line.match(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}/);
+  if (!timestampMatch) return line;
+
+  const timestamp = new Date(timestampMatch[0]);
+  const content = line.slice(timestampMatch[0].length + 1);
+
+  return (
+    <div key={line} className="flex">
+      <span className="text-gray-400 mr-4 select-none whitespace-nowrap shrink-0">
+        {timestamp.toLocaleTimeString()}
+      </span>
+      <span className="break-all">{content}</span>
+    </div>
+  );
+};
+
+const LogDisplay = ({
+  logs,
+  isLoading,
+}: {
+  logs: {
+    log_text: string;
+  };
+  isLoading: boolean;
+}) => {
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!logs || !logs.log_text) {
+    return <div>No logs available</div>;
+  }
+
+  const lines = logs.log_text.split("\n");
+  return (
+    <div className="space-y-1">
+      {lines.map((line: string, index: number) => (
+        <div key={index}>{formatLogLine(line)}</div>
+      ))}
+    </div>
+  );
+};
+
+const ExecutionDetailsSheetInner = () => {
+  const { id: executionId } = runDetailRoute.useParams() as RunDetailParams;
```

Comment:
! The execution ID needs to be the child execution ID, not the root graph execution ID.